Dockerfile

```
@@ -1,4 +1,34 @@
-FROM node:20-alpine AS builder
+# Multi-stage Dockerfile that supports both development and production builds
+ARG NODE_VERSION=22
+
+# Development target
+FROM node:${NODE_VERSION} AS development
+
+# Install system dependencies for development
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)